src/core/tests/policy_tests.rs

```diff
@@ -71,3 +71,27 @@ fn policy_group_from_java() {
         .unwrap();
     assert_eq!(vec![10, 6], dpt_hr.clone().into_vec());
 }
+
+#[test]
+fn parse_boolean_expression() {
+    let access_policy = AccessPolicy::from_boolean_expression(
+        "(Department::HR | Department::RnD) & Level::level_2",
+    )
+    .unwrap();
+    let expected_ap = (ap("Department", "HR") | ap("Department", "RnD")) & ap("Level", "level_2");
```

Comment:
is it commutative? is the following expected to work? if yes, please add this test

"Level::level_2 & (Department::HR | Department::RnD)"

Reply:
done